rpc/amqp-rpc.go

```diff
@@ -494,39 +475,36 @@ func (rpc *AmqpRPCServer) catchSignals() {
 // continue blocking until it has processed any messages that have already been
 // retrieved.
 func (rpc *AmqpRPCServer) Stop() {
+	rpc.mu.Lock()
+	rpc.done = true
+	rpc.mu.Unlock()
 	if rpc.connected {
```

Comment:
So, this isn't your bug, but I think connected needs to be under a lock given how it's used elsewhere. We can do this in a follow up.